opal/corelib/module.rb

```diff
@@ -615,9 +641,16 @@ def name
 
   def remove_const(name)
     %x{
-      var old = self.$$scope[name];
-      delete self.$$scope[name];
-      return old;
+      if (self.$$const[name] != null) {
+        var old = self.$$const[name];
+        delete self.$$const[name];
+        return old;
+      } else if (self.$$autoload != null && self.$$autoload[name] != null) {
+        delete self.$$autoload[name];
+        return nil;
+      } else {
+        #{raise NameError, "constant #{self}::#{name} not defined"}
+      }
     }
```

Comment:
All of this would probably look better if you removed the `else if { } else { }` since you're returning from each of them.